From: Keith Busch <kbusch@kernel.org>
Date: Mon, 30 Oct 2023 08:13:09 -0700
Subject: nvme: introduce helper function to get ctrl state
Patch-mainline: v6.7-rc5
Git-commit: 5c687c287c46fadb14644091823298875a5216aa
References: jsc#PED-6252 jsc#PED-5728 jsc#PED-5062 jsc#PED-3535

The controller state is typically written by another CPU, so reading it
should ensure no optimizations are taken. This is a repeated pattern in
the driver, so start with adding a convenience function that returns the
controller state with READ_ONCE().

Reviewed-by: Sagi Grimberg <sagi@grimberg.me>
Signed-off-by: Keith Busch <kbusch@kernel.org>
Acked-by: Daniel Wagner <dwagner@suse.de>
---
 drivers/nvme/host/nvme.h |    5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/nvme/host/nvme.h
+++ b/drivers/nvme/host/nvme.h
@@ -387,6 +387,11 @@ struct nvme_ctrl {
 	enum nvme_dctype dctype;
 };
 
+static inline enum nvme_ctrl_state nvme_ctrl_state(struct nvme_ctrl *ctrl)
+{
+	return READ_ONCE(ctrl->state);
+}
+
 enum nvme_iopolicy {
 	NVME_IOPOLICY_NUMA,
 	NVME_IOPOLICY_RR,
